node/mock-test.ts

```diff
@@ -125,4 +129,176 @@ export class MockTestRunner {
             console.log('TRACE FILE: ' + traceFile);
         }
     }
+
+    // Returns a path to node.exe with the correct version for this task (based on if its node10 or node)
+    private getNodePath(nodeVersion?: number): string {
+        const version: number = nodeVersion || this.getNodeVersion();
+
+        let downloadVersion: string;
+        switch (version) {
+            case 5:
+                downloadVersion = 'v5.10.1';
+                break;
+            case 6:
+                downloadVersion = 'v6.10.3';
+                break;
+            case 10:
+                downloadVersion = 'v10.15.1';
+                break;
+            default:
+                throw new Error('Invalid node version, must be 5, 6, or 10 (received ' + version + ')');
+        }
+
+        // Install node in home directory if it isn't already there.
+        const downloadDestination: string = path.join(downloadDirectory, 'node' + version);
+        const pathToExe: string = this.getPathToNodeExe(downloadVersion, downloadDestination);
+        if (pathToExe) {
+            return pathToExe;
+        }
+        else {
+            return this.downloadNode(downloadVersion, downloadDestination);
```

Comment:
do we need to add node to the PATH?

Reply:
No, as we discussed offline this mimics the agent behavior.